frontend/ui/uimanager.lua

```diff
@@ -57,33 +57,23 @@ function UIManager:init()
             self:_startAutoSuspend()
         end
         self.event_handlers["PowerPress"] = function()
-            self._power_ev_handled = false
-            local showPowerOffDialog = function()
-                if self._power_ev_handled then return end
-                self._power_ev_handled = true
-                local ConfirmBox = require("ui/widget/confirmbox")
-                UIManager:show(ConfirmBox:new{
-                    text = _("Power off?"),
-                    ok_callback = function()
-                        local InfoMessage = require("ui/widget/infomessage")
-
-                        UIManager:show(InfoMessage:new{
-                            text = _("Powered off."),
-                        })
-                        -- The message can fail to render if this is executed directly
-                        UIManager:scheduleIn(0.1, function()
-                            self:broadcastEvent(Event:new("Close"))
-                            Device:powerOff()
-                        end)
-                    end,
-                })
+            self.poweroff_action = function()
```

Comment:
Does this function need to be created every time power button is pressed? If not, we can declare it as a method for `UIManager`.